ir/metadata/specialized_metadata.go

```diff
@@ -1429,7 +1429,9 @@ type DIModule struct {
 	Name         string // required.
 	ConfigMacros string // optional; empty if not present.
 	IncludePath  string // optional; empty if not present.
-	Sysroot      string // optional; empty if not present.
+	APINotes     string // optional; empty if not present.
+	File         Field  // optional; empty if not present.
+	Line         *int64 // optional; empty if not present.
```

Comment:
As Go is lacking optional values, using pointers is a common solution.

For the specialized metadata nodes, I've tried to rely on zero values for omitted values, where those zero values give no information regarding the source. For instance, line number 0 does not exist, as source line numbers are 1-indexed. For this reason, we can use a regular zero value of `int64` for the Line field.

This is how it was done for `DICommonBlock`:

From https://github.com/llir/llvm/blob/fbfa5b12032936ac00ed05780808a6f917694f78/ir/metadata/specialized_metadata.go#L102
```go
type DICommonBlock struct {
	...
	Line        int64   // optional; zero value if not present.
}
```

The balance here is between correctness and ease of use. Having an API with pointer values for every field makes it more difficult to use.

As both approaches have merit, we should decide on one approach and use it throughout for consistency. So either we change `Line` to be `int64` to be consistent with `DICommonBlock` (and others). Or we change all other places to use pointers as well.

Definitely open for input.

Cheers,
Robin